app/Models/File.php

```diff
@@ -154,15 +154,21 @@ public function getInfo()
      *
      * @param array $tags  The (selective) tags to sync (if the song exists)
      * @param bool  $force Whether to force syncing, even if the file is unchanged
+     * @param bool  $path  Whether to force syncing of a specific path, even
+     *                     if the file is unchanged
      *
      * @return bool|Song A Song object on success,
      *                   true if file exists but is unmodified,
      *                   or false on an error.
      */
-    public function sync($tags, $force = false)
+    public function sync($tags, $force = false, $path = null)
```

Comment:
So what data type is `$path` exactly? The method doc says `bool`, the default value is `null`, and you run some `preg_match` on it??